### PR TITLE
[DON'T MERGE] CCIP Repository Merge - ocr2

### PR DESCRIPTION
**Pull request by asoliman92**
**DO NOT MERGE THAT PR**, once approved and working we need to figure out way to merge it instead of squashing. Otherwise, entire effort on preserving commits history from CCIP repo will be lost.

Next steps:
- [ ] ccip eng review
- [ ] core eng review
- [ ] figuring out strategy for merge instead of squash

### CODE REVIEW

File: core/services/feeds/service.go [hunk 1]
```diff
@@ -107,6 +107,9 @@ type Service interface {
 	ListSpecsByJobProposalIDs(ctx context.Context, ids []int64) ([]JobProposalSpec, error)
 	RejectSpec(ctx context.Context, id int64) error
 	UpdateSpecDefinition(ctx context.Context, id int64, spec string) error
+
+	// UnsafeSetConnectionsManager Only for testing
+	UnsafeSetConnectionsManager(ConnectionsManager)
```
Comment (mateusz-sekara) [inline core/services/feeds/service.go:112]: @asoliman92 Have you checked if it's still needed? Locally `integration_tests.go` and `integration_legacy_test.go` works for me when removed

Checking here https://github.com/smartcontractkit/ccip/pull/1253

Comment (mateusz-sekara) [reply core/services/feeds/service.go:112]: Ignore that, it's needed for CLO tests I created long time ago :P (Test_CLOSpecApprovalFlow)

File: core/services/ocr2/delegate.go [hunk 2]
```diff
@@ -305,6 +316,51 @@ func (d *Delegate) cleanupEVM(ctx context.Context, jb job.Job, relayID types.Rel
...
+
+		srcProvider, _, err2 := d.ccipExecGetSrcProvider(ctx, jb, pluginJobSpecConfig, transmitterID, dstProvider)
+		if err2 != nil {
+			return err
+		}
+		err2 = ccipexec.UnregisterExecPluginLpFilters(srcProvider, dstProvider)
+		if err2 != nil {
+			d.lggr.Errorw("failed to unregister ccip exec plugin filters", "err", err2, "spec", spec)
+		}
+		return nil
```
Comment (jmank88) [inline core/services/ocr2/delegate.go:363]: Why does this logic live in `cleanupEVM` as opposed to triggering when a service is closed?

File: core/services/relay/evm/exec_provider.go [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+	unregisterFuncs = append(unregisterFuncs, func() error {
+		if s.usdcAttestationAPI == "" {
+			return nil
+		}
+		return ccip.CloseUSDCReader(s.lggr, s.lggr.Name(), s.usdcSrcMsgTransmitterAddr, s.lp)
+	})
+	var multiErr error
+	for _, fn := range unregisterFuncs {
+		if err := fn(); err != nil {
+			multiErr = multierr.Append(multiErr, err)
```
Comment (jmank88) [inline core/services/relay/evm/exec_provider.go:122]: We've been trying to drop this import in favor of the std lib:
```suggestion
			multiErr = errors.Join(multiErr, err)
```
That can be a follow up change though.

File: core/services/relay/evm/statuschecker/txm_status_checker.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+		allStatuses = append(allStatuses, status)
+		counter++
+
+		// Break the loop if the cap is reached
+		if counter >= maxStatuses {
+			return allStatuses, counter - 1, fmt.Errorf("maximum number of statuses reached, possible infinite loop")
+		}
+	}
+
+	return allStatuses, counter - 1, nil
```
Comment (jmank88) [inline core/services/relay/evm/statuschecker/txm_status_checker.go:53]: Why subtract one? Returning from here means the break on 42 was encountered, _before_ incrementing on 45. Isn't `counter` accurate as-is?

**Comment by winder** [comment]
To review this we should start with the following:
1. A script/algorithm for how the files were moved to chainlink. What was the original commit on the ccip repo?
2. A script to diff the source ccip commit files and the dest chainlink files.
3. Targeted review on the diff between source and dest.

On a quick inspection it looks like about 1/3 of the files were modified during the migration:
```
#!/usr/bin/env bash

if [[ "$OSTYPE" == "darwin"* ]]; then
    # macOS
    SUM=md5
else
    # Assume Linux
    SUM=md5sum
fi

pushd ccip/core/services/ocr2/plugins/ccip/ > /dev/null
echo -e "ccip repo commit: \n$(git log --max-count=1 --oneline --decorate)"
find . -type f -exec $SUM {} \;  > ../files.md5
popd > /dev/null
mv ./ccip/core/services/ocr2/plugins/files.md5 ./chainlink/core/services/ocr2/plugins/files.md5
pushd ./chainlink/core/services/ocr2/plugins/ccip/ > /dev/null
echo -e "chainlink repo commit: \n$(git log --max-count=1 --oneline --decorate)"

echo ""

echo "Source files not modified:"
$SUM -c ../files.md5 2>/dev/null | grep OK | wc -l
echo "Source files modified:"
$SUM -c ../files.md5 2>/dev/null | grep -v OK | wc -l
```

```
ccip repo commit:
1da6ea640d (HEAD -> ccip-develop) Cleanup http test servers (#1163)
chainlink repo commit:
01a5830252 (HEAD -> ccip/merge-ocr2, origin/ccip/merge-ocr2) Adding mockery configuration for ccip specific code

Source files not modified:
125
Source files modified:
60
```

**Review by mateusz-sekara — APPROVED** [review]

**Comment by winder** [comment]
The following commits are identical according to [the script](https://github.com/smartcontractkit/chainlink/pull/13975#issuecomment-2265320336):

ccip: `git checkout 44b23a5a32`
chainlink: `git checkout ccip/merge-ocr2 && git checkout b851bb3e87`

**Pull request closed without merging**